crates/storage/provider/src/providers/database.rs

```diff
@@ -208,14 +235,13 @@ impl<DB: Database> BlockProvider for ShareableDatabase<DB> {
     fn block(&self, id: BlockHashOrNumber) -> Result<Option<Block>> {
         let tx = self.db.tx()?;
         if let Some(number) = convert_hash_or_number(&tx, id)? {
-            if let Some(header) = self.header_by_number(number)? {
```

Comment:
I see that you save perf here, but just making the maybe obvious comment that it only benefits where you had calls on shareable db, whereas some of the changes are for places where we already had the TX